lms/resources/lti_launch/factory.py

```diff
@@ -0,0 +1,37 @@
+from lms.resources.lti_launch.base import LTILaunchResource
+from lms.resources.lti_launch.canvas import CanvasLTILaunchResource
+from lms.resources.lti_launch.blackboard import BlackboardLTILaunchResource
+
+from lms.models import ApplicationInstance
+
+
+def _is_canvas(request):
```

Comment:
We'll eventually might have to use this in multiple places.

I think the less ugly way of doing this having 

`def ApplicationInstanceService.product(self, request)`

Sometimes we can decide (or we thinkg we can) solely with the model's product but the request might be needed like in canvas.